### PR TITLE
Remove forwarding of matched orders

### PR DESCRIPTION
**Pull request by sporejack**
# Motivation
The new API architecture calls for no communication between the OME and the Executioner.

# Changes
 - Remove the `rpc` module
 - Remove instances of dependencies to the `rpc` module

### CODE REVIEW

File: src/book.rs [hunk 1]
```diff
@@ -184,7 +183,7 @@ impl Book {
     async fn r#match(
         &mut self,
         mut order: Order,
-        executioner_address: String,
+        _executioner_address: String,
```
Comment (chrdevmar) [inline src/book.rs:186]: can this just be removed?

Comment (kumar-ish) [inline src/book.rs:186]: Isn't this parameter redundant now?

**Review by chrdevmar — CHANGES_REQUESTED** [review]
I expected to see changes to the response to include the details of the match. Is that supposed to be in here?

File: src/book.rs (hunk 1 above)
Comment (sporejack) [reply src/book.rs:186]: Yes. I can drop it in this PR if we want, but I was going to separate it out (note the base branch).

Comment (sporejack) [reply src/book.rs:186]: Yes, see my above response to @kumar-ish

**Comment by sporejack** [comment]
> Is that supposed to be in here?

No, I was going to separate out the changesets for the new API. My interpretation of the scope of the associated ticket (734) was to remove the order forwarding.

Note the base branch for this and #57 is the staging branch for the new API architecture -- there will be no braking changes to production builds.

**Review by chrdevmar — APPROVED** [review]

File: src/book.rs (hunk 1 above)
Comment (kumar-ish) [reply src/book.rs:186]: :rocket:

**Pull request merged**